### PR TITLE
update cli to support arc gateway and workload identity

### PR DESCRIPTION
**Pull request by bavneetsingh16**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/connectedk8s/azext_connectedk8s/custom.py [hunk 1]
```diff
@@ -498,58 +500,79 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
...
+    # Get helm chart path
+    chart_path = utils.get_chart_path(registry_path, kube_config, kube_context, helm_client_location)
+
+    helm_content_values = helm_values_dp["helmValuesContent"]
+
+    # Substitute any protected helm values as the value for that will be null
+    for helm_parameter, helm_value in protected_helm_values.items():
+        helm_content_values[helm_parameter] = helm_value
+
+    print("Starting to install Azure arc agents on the Kubernetes cluster.")
```
Comment (deeksha345) [inline src/connectedk8s/azext_connectedk8s/custom.py:539]: Can you please update to the new log type "print("Step: {}: ...")

File: src/connectedk8s/azext_connectedk8s/custom.py [hunk 2]
```diff
@@ -498,58 +500,79 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
...
     while True:
         connected_cluster = get_connectedk8s_2024_07_01(cmd, resource_group_name, cluster_name)
-        # Checking for provisioning state now as agent state feedback loop is not implemented
-        if connected_cluster.provisioning_state == "Succeeded" or connected_cluster.provisioning_state == "Failed":
-            print("Agent state has reached terminal state of: " + connected_cluster.provisioning_state)
+        if connected_cluster.arc_agent_profile.agent_state == consts.Agent_State_Succeeded or \
+            connected_cluster.arc_agent_profile.agent_state == consts.Agent_State_Failed:
             return True
         elapsed_time = time.time() - start_time
         if elapsed_time >= timeout_minutes * 60:
```
Comment (deeksha345) [inline src/connectedk8s/azext_connectedk8s/custom.py:578]: Can you please remove the print statement after this if, I forgot to do so when merging in my changes.

File: src/connectedk8s/azext_connectedk8s/custom.py (hunk 1 above)
Comment (bavneetsingh16) [reply src/connectedk8s/azext_connectedk8s/custom.py:539]: done

File: src/connectedk8s/azext_connectedk8s/custom.py (hunk 2 above)
Comment (bavneetsingh16) [reply src/connectedk8s/azext_connectedk8s/custom.py:578]: done

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by bavneetsingh16** [comment]
> Please fix CI issues

For azdev linter, I ran the cmd locally and it passes:
image


For azdev style, these are all the changes that pre existed.

**Comment by zhoxing-ms** [comment]
@yanzhudd Could you please help review this PR when you have time?

File: src/connectedk8s/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,13 @@
 Release History
 ===============
 
+1.10.0
+++++++
+* Added support for arc gateway feature.
```
Comment (yanzhudd) [inline src/connectedk8s/HISTORY.rst:8]:
```suggestion
* Add support for arc gateway feature.
```

**Comment by yanzhudd** [comment]
please note Azure CLI team is only responsible for code style and specification but not business logic, so it is better to involve service team to review this PR as well.

**Comment by yanzhudd** [comment]
please let me confirm with you if there are no tests added in this PR for the changes. If so, please add tests to verify if the changes meet your expectation.

File: src/connectedk8s/HISTORY.rst (hunk 3 above)
Comment (yanzhudd) [reply src/connectedk8s/HISTORY.rst:8]: it is better to specify which commands/parameters are impacted

Comment (bavneetsingh16) [reply src/connectedk8s/HISTORY.rst]: done, thanks for reviewing!

**Review by deeksha345 — APPROVED** [review]

**Comment by bavneetsingh16** [comment]
@yanzhudd, Deeksha and I are members of the service team owning this CLI extension. I have made the necessary updates, and we have thoroughly reviewed and extensively tested them, including a bug bash session. Additionally, we utilized the custom .whl file for this version within our testing framework, which allows us to test across all regions, providing greater confidence. This testing covered all scenarios for the CLI extension and the newly introduced gateway changes.
 
Could you please review the changes again and merge if everything looks good?

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**